File: app/src/main/java/com/tokopedia/tkpd/recharge/view/widget/WrapContentViewPager.java
package com.tokopedia.tkpd.recharge.view.widget;

import android.content.Context;
import android.support.v4.view.ViewPager;
import android.util.AttributeSet;
import android.view.View;

/**
 *
 * Created by Tokopedia 05 on 7/11/2016.
 */
public class WrapContentViewPager extends ViewPager {
    private View currentView;

    public WrapContentViewPager(Context context) {
        super(context);
    }

    public WrapContentViewPager(Context context, AttributeSet attrs) {
        super(context, attrs);
    }

    @Override
    protected void onMeasure(int widthMeasureSpec, int heightMeasureSpec) {

        if (currentView == null) {
            super.onMeasure(widthMeasureSpec, heightMeasureSpec);
            return;
        }
        int height = 0;
        currentView.measure(widthMeasureSpec, MeasureSpec.makeMeasureSpec(0, MeasureSpec.UNSPECIFIED));
        int h = currentView.getMeasuredHeight();
        if (h > height) height = h;
        heightMeasureSpec = MeasureSpec.makeMeasureSpec(height, MeasureSpec.EXACTLY);


        super.onMeasure(widthMeasureSpec, heightMeasureSpec);
    }
    public void measureCurrentView(View currentView) {
        this.currentView = currentView;
        requestLayout();
    }
}

File: features/content/feed_component/src/main/java/com/tokopedia/feedcomponent/view/widget/WrapContentViewPager.kt
package com.tokopedia.feedcomponent.view.widget

import android.content.Context
import android.util.AttributeSet
import android.view.View
import com.tokopedia.abstraction.base.view.widget.TouchViewPager

/**
 * @author by milhamj on 17/12/18.
 */
class WrapContentViewPager : TouchViewPager {

    private var currentView: View? = null

    constructor(context: Context?) : super(context)
    constructor(context: Context?, attrs: AttributeSet?) : super(context, attrs)

    override fun onMeasure(widthMeasureSpec: Int, heightMeasureSpec: Int) {
        var modifiedHeight = heightMeasureSpec
        if (currentView == null) {
            super.onMeasure(widthMeasureSpec, modifiedHeight)
            return
        }
        var height = 0
        currentView?.measure(widthMeasureSpec, View.MeasureSpec.makeMeasureSpec(0, View.MeasureSpec.UNSPECIFIED))
        val h = currentView?.measuredHeight ?: 0
        if (h > height) height = h
        modifiedHeight = View.MeasureSpec.makeMeasureSpec(height, View.MeasureSpec.EXACTLY)

        super.onMeasure(widthMeasureSpec, modifiedHeight)
    }

    fun measureCurrentView(currentView: View) {
        this.currentView = currentView
        requestLayout()
    }
}